### PR TITLE
Organize repo

### PR DESCRIPTION
**Pull request by jaberkow**
Hi Nina,

This pull request does three things:

1.  I fixed a few updates to "finite_diff_wave.py" that got lost on another branch.
2.  I made major refactor to the code organization that I hope looks cleaner.  I have a folder 'src/' that holds the python scripts for training, rolling out, and visualizing and has a subfolder 'environments/' that holds the code for the open ai gym environments.  All the python scripts can be called from the top level of the project folder.
3.  I added a test script that makes sure the config file has allowable values.

Let me know if you have any questions.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -50,42 +55,49 @@ Install the packages from `requirements.txt`
 ```
 $ pip install -r requirements.txt
 ```
+### Tests
+
+If any changes are made to the default values in `configs/config.yml` , run the following command
+
+```
+python tests/config_test.py
```
Comment (ninalopatina) [inline README.md:63]: Great that you added a test here. I ran the test and got this output: 

YAMLLoadWarning: calling yaml.load() without Loader=... is deprecated, as the default Loader is unsafe. Please read https://msg.pyyaml.org/load for full details.
  cfg = yaml.load(ymlfile)
..
----------------------------------------------------------------------
Ran 2 tests in 0.008s

OK
_____
It works but the warning should be addressed or suppressed

File: README.md [hunk 2]
```diff
@@ -50,42 +55,49 @@ Install the packages from `requirements.txt`
...
 
 ## About the environment
 
 This package simulates an oscillating bridge by modelling it with the one-dimensional [wave equation](https://en.wikipedia.org/wiki/Wave_equation), which is simulated using a [finite difference solver](https://en.wikipedia.org/wiki/Finite_difference_method).  The action space of the environment represents pistons that apply a force to actively dampen vibrations in the bridge.  The reward signal is inversely proportional to the interal energy of the system.  One episode of the environment involves 3 phases:  1) A "warmup phase" where an external force is applied to the system to cause oscillations 2) An "equilibriation" phase where the oscillations settle in to stable patterns and 3) A dampening phase where the agent attempts to dampen the oscillations.
 
 ## Training an agent
 
-To train an agent for 1000 timesteps on the vibrating bridge environment and save it as `my_first_agent.pkl`, navigate to the `complex_envs` folder and run the following command:
+To train an agent for 40,000 timesteps on the vibrating bridge environment and save it as `trained_agents/my_first_agent.pkl`, run the following command:
```
Comment (ninalopatina) [inline README.md:73]: Good increase on the training timesteps. I get better results here, but I also get these deprecation messages, which should either be addressed or suppressed:

src/train.py:48: YAMLLoadWarning: calling yaml.load() without Loader=... is deprecated, as the default Loader is unsafe. Please read https://msg.pyyaml.org/load for full details.
  cfg = yaml.load(ymlfile)
WARNING: Logging before flag parsing goes to stderr.
W0702 15:52:06.551560 4622173632 deprecation_wrapper.py:119] From /anaconda3/envs/WaveEnv/lib/python3.6/site-packages/stable_baselines/common/tf_util.py:98: The name tf.ConfigProto is deprecated. Please use tf.compat.v1.ConfigProto instead.

W0702 15:52:06.551992 4622173632 deprecation_wrapper.py:119] From /anaconda3/envs/WaveEnv/lib/python3.6/site-packages/stable_baselines/common/tf_util.py:107: The name tf.Session is deprecated. Please use tf.compat.v1.Session instead.

W0702 15:52:06.557882 4622173632 deprecation_wrapper.py:119] From /anaconda3/envs/WaveEnv/lib/python3.6/site-packages/stable_baselines/common/policies.py:114: The name tf.variable_scope is deprecated. Please use tf.compat.v1.variable_scope instead.

W0702 15:52:06.558548 4622173632 deprecation_wrapper.py:119] From /anaconda3/envs/WaveEnv/lib/python3.6/site-packages/stable_baselines/common/input.py:25: The name tf.placeholder is deprecated. Please use tf.compat.v1.placeholder instead.

W0702 15:52:06.565917 4622173632 deprecation.py:323] From /anaconda3/envs/WaveEnv/lib/python3.6/site-packages/stable_baselines/common/policies.py:561: flatten (from tensorflow.python.layers.core) is deprecated and will be removed in a future version.
Instructions for updating:
Use keras.layers.flatten instead.
W0702 15:52:06.810468 4622173632 deprecation_wrapper.py:119] From /anaconda3/envs/WaveEnv/lib/python3.6/site-packages/stable_baselines/a2c/utils.py:156: The name tf.get_variable is deprecated. Please use tf.compat.v1.get_variable instead.

W0702 15:52:07.068109 4622173632 deprecation_wrapper.py:119] From /anaconda3/envs/WaveEnv/lib/python3.6/site-packages/stable_baselines/ppo2/ppo2.py:193: The name tf.summary.scalar is deprecated. Please use tf.compat.v1.summary.scalar instead.

W0702 15:52:07.097346 4622173632 deprecation.py:323] From /anaconda3/envs/WaveEnv/lib/python3.6/site-packages/tensorflow/python/ops/math_grad.py:1250: add_dispatch_support.<locals>.wrapper (from tensorflow.python.ops.array_ops) is deprecated and will be removed in a future version.
Instructions for updating:
Use tf.where in 2.0, which has the same broadcast rule as np.where
W0702 15:52:07.349545 4622173632 deprecation_wrapper.py:119] From /anaconda3/envs/WaveEnv/lib/python3.6/site-packages/stable_baselines/ppo2/ppo2.py:209: The name tf.train.AdamOptimizer is deprecated. Please use tf.compat.v1.train.AdamOptimizer instead.

W0702 15:52:07.613144 4622173632 deprecation_wrapper.py:119] From /anaconda3/envs/WaveEnv/lib/python3.6/site-packages/stable_baselines/ppo2/ppo2.py:245: The name tf.summary.merge_all is deprecated. Please use tf.compat.v1.summary.merge_all instead.

W0702 15:52:07.614977 4622173632 deprecation_wrapper.py:119] From /anaconda3/envs/WaveEnv/lib/python3.6/site-packages/stable_baselines/common/base_class.py:830: The name tf.summary.FileWriter is deprecated. Please use tf.compat.v1.summary.FileWriter instead.

Comment (ninalopatina) [inline README.md:73]: file shoudl have a more informative name than "my first agent"

File: README.md [hunk 3]
```diff
@@ -50,42 +55,49 @@ Install the packages from `requirements.txt`
...
-To visualize a rollout saved in `complex_envs/output.npz`, run the following command from the same folder:
+To visualize a rollout saved in `rollouts/output.npz`, run the following command:
 
 ```
-# python visualize.py -i output.npz -f first_output
+# python src/visualize.py -i rollouts/output.npz -f rollouts/first_output
 ```
 
-This will produce two files `first_output.eps` which plots the trajectory of the energy over the episode and `first_output.gif` which is an animation of the bridge and the impulse force.
+This will produce two files `first_output.png` which plots the trajectory of the energy over the episode and `first_output.gif` which is an animation of the bridge and the impulse force.
```
Comment (ninalopatina) [inline README.md:101]: rollout file should have a more informative name than first_output

File: README.md [hunk 4]
```diff
@@ -1,15 +1,20 @@
...
-	* active_damping_env.py : This is a class definition for an OpenAI gym environment simulating an oscillating bridge
+	* environments/ : This folder contains code for the environments.
+		* finite_diff_wave.py : This is a class definition for a simulator of one dimensional wave equation with finite difference methods.
+		* active_damping_env.py : This is a class definition for an OpenAI gym environment simulating an oscillating bridge
 * configs/ 
 	* config.yml : This file holds the default parameters for the scripts and environments
+* tests/
+	* config_test.py :  A unnittest test fixture that can be used to make sure `configs/config.yml` has all the appropriate keys and valid parameter settings
+* trained_agents/ : A folder for storing trained agents
+* rollouts/ : A folder for storing rollouts of trained agents
```
Comment (ninalopatina) [inline README.md:17]: It would be good to add sample .gif, .png, and .npz files here so users can see what the output is supposed to be, and for users to be able to take a quick look at the result without having to install and run everything themselves

File: README.md [hunk 5]
```diff
@@ -50,42 +55,49 @@ Install the packages from `requirements.txt`
...
+The rollout will be saved as `rollouts/output.npz`, which can be changed by passing `-f <filename>` to the above command.  Note that the trajectories produced will have length equal to the number of rollout steps + the number of warmup steps + the number of equilibriation steps (values of which are set in `/configs/config.yml`).
 
 ### Visualizing a rollout
 
-To visualize a rollout saved in `complex_envs/output.npz`, run the following command from the same folder:
+To visualize a rollout saved in `rollouts/output.npz`, run the following command:
 
 ```
-# python visualize.py -i output.npz -f first_output
+# python src/visualize.py -i rollouts/output.npz -f rollouts/first_output
```
Comment (ninalopatina) [inline README.md:98]: Would be good to add another line for user to open the file from command line, or incorporate that into the python script visualize.py

File: README.md [hunk 6]
```diff
@@ -50,42 +55,49 @@ Install the packages from `requirements.txt`
...
 ```
```
Comment (ninalopatina) [inline README.md:99]: Toward the end of the readme, it would be best to include a note about which variables users should experiment with if they were to run your repo: highlight 1 or 2, describe what they do, and what the bounds are for changing them

**Review by ninalopatina — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (jaberkow) [reply README.md:63]: Addressed.

File: README.md (hunk 5 above)
Comment (jaberkow) [reply README.md:98]: Added command line instructions.
